### DISCUSSION TITLE
Cannot select a schema (database) if not informed at connection time

### DISCUSSION THREAD

**Discussion by mlopezgva** [tconbeer/harlequin#508] [Q&A, Troubleshooting, Support] [answered]
Hello!

I am rynig Harlequin as a one step beyon mycli and mypsql enhanced clients... And it looks quite good.

HOWEVER... I work with several databases, each with several schemas.

I set up a connection using the host, port user etc., but NO "database", as I use three schemas for a single project (which consists on three separated but intertwined apps). So, I can see the schemas, which it's great. But, when I try to dosomething like 

```sql
SELECT * FROM config WHERE updated_at > CURRENT_DATE - 7 DAY
```

Y get this:

    1046 (3D000): No database selected

I tried to ENTER, CTRL+ENTER, SHIFT*ENTER, ALT+ENTER and many other keybindings... None allows me to "select active schema", which forces me to prepend the schema to the table name... like `backendapp.config`.

Also, if at least hitting CTR+ENTER / CLICK on the table name would copy the schema, at least it would be easier... But still I'll have to edit any query I paste to the editor.

Any chance to get a new functionality to "Change active Schema" (and show it in different color, or in the editor's legend).

Thanks!

**Comment by tconbeer** [comment, marked as answer]
It's interesting to think about what a cross-database interface would look like for this; Snowflake has something similar in the snowsight UI. This might be more of an adapter-specific feature, in which case I think the interface would be through a right-click on the data catalog (see https://github.com/tconbeer/harlequin/issues/213, where I already anticipated this). 

I think you're using the mysql adapter, but that's not totally clear. 

For now... can't you use a `USE database` statement before your select query? I think the two queries get executed in the same session, so the `USE` statement should work:

```sql
USE backendapp;
SELECT * FROM config WHERE updated_at > CURRENT_DATE - 7 DAY
```

**Reply by mlopezgva** [reply]
Hi!

Sorry, but no. I thought it should work, too, but it does not. Hence my suggestion. Something like Ctrl-click or a right-click menu, etc. Several options:

Alt-S (Select database)
Ctrl-click
Ctrl-U ("U"se database)
...

Also, double-click copies the schema / table / column name... Ctrl+double-click, or Shift+double-click could copy a fully qualified name, something like

- database_One
- database_Two         # <kbd>DblClick</kbd> here writes `database_Two` in the editor
  - table_1                   # <kbd>DblClick</kbd> copies `table_1`
  - table_2                   # <kbd>Shift+DblClick</kbd> copies `database_two.table_1`
    - table_id                # <kbd>DblClick</kbd>  copies `table_id`, <kbd>Shift+DblClick</kbd> copies`database_two.table_2.table_id` (or reverse... but I saw DblClick / Ctrl|Shift+Enter **already copies 'database.table', good for now**)
    - column_one
    - column_two
  - table 3
- database 3

Something like that. This can also be independent of the "use" case. :-)

**Reply by tconbeer** [reply]
created this issue, since `use database` should really work: https://github.com/tconbeer/harlequin-mysql/issues/11

I think the feature request is already captured in #213, so I'm going to close this discussion for now.

**Reply by tconbeer** [reply]
@mlopezgva This has been patched in harlequin-mysql v0.2.0. Please upgrade for the fix.
